src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImageFunctionalTest.groovy

```diff
@@ -369,4 +379,31 @@ class DockerBuildImageFunctionalTest extends AbstractGroovyDslFunctionalTest {
             }
         """
     }
+
+    private static String buildMultiStageImage() {
+        """
+            import com.bmuschko.gradle.docker.tasks.image.Dockerfile
+            import com.bmuschko.gradle.docker.tasks.image.DockerBuildImage
+            import com.bmuschko.gradle.docker.tasks.image.DockerRemoveImage
+
+            task dockerfile(type: Dockerfile) {
+                from '$TEST_IMAGE_WITH_TAG', 'stage1'
+                label(['maintainer': 'stage1 - ${UUID.randomUUID().toString()}'])
+
+                from '$TEST_IMAGE_WITH_TAG', 'stage2'
+                label(['maintainer': 'stage2 - ${UUID.randomUUID().toString()}'])
+            }
+
+            task buildTarget(type: DockerBuildImage) {
+                dependsOn dockerfile
+                target = "stage2"
+            }
+            
+            task removeImage(type: DockerRemoveImage) {
+                dependsOn buildTarget
```

Comment:
We'll need to restructure this a bit. The test should execute `buildTarget`. The task `buildTarget` should use `finalizedBy removeImage` and remove `dependsOn buildTarget`. In its current state `removeTarget` would not be executed if `buildTarget` fails.